pkg/api/api.go

```diff
@@ -87,13 +93,14 @@ func (cfg *Config) wrapDistributorPush(d *distributor.Distributor) push.Func {
 }
 
 type API struct {
-	AuthMiddleware middleware.Interface
-
-	cfg       Config
-	server    *server.Server
-	logger    log.Logger
-	sourceIPs *middleware.SourceIPExtractor
-	indexPage *IndexPageContent
+	AuthMiddleware       middleware.Interface
+	cfg                  Config
+	server               *server.Server
+	logger               log.Logger
+	sourceIPs            *middleware.SourceIPExtractor
+	indexPage            *IndexPageContent
+	HTTPHeaderMiddleware HTTPHeaderMiddleware
+	LogHeaders           bool
```

Comment:
Do we need this here? Cant we just have the HTTPHeaderMiddleware and in the checks do 
`
if a.HTTPHeaderMiddleware != nil
`